tracer/src/Datadog.Trace/Logging/TracerFlare/DebugLogReader.cs

```diff
@@ -65,7 +67,12 @@ public static async Task WriteDebugLogArchiveToStream(Stream writeTo, string log
         try
         {
             using var monitoringStream = new WriteCountingStream(writeTo);
-            using var archive = new ZipArchive(monitoringStream, ZipArchiveMode.Create, true);
+            using var archive = new Vendors.ICSharpCode.SharpZipLib.Zip.ZipOutputStream(monitoringStream);
+            archive.IsStreamOwner = false; // Do not close the underlying stream when closing the archive
+            archive.SetLevel(9); // Set to optimal compression
+
+            // Create a fixed-size buffer to bound memory operations
+            byte[] buffer = new byte[4096];
```

Comment:
Given that the changes are all restricted to this file, I think it should be easy enough to differentiate?

Reply:
Implemented in 59971aa153eadac7fd3f9dcf0cfc51b3a79da14c